pulsar/impl_message.go

```diff
@@ -184,7 +184,7 @@ func (id *messageID) BatchSize() int32 {
 }
 
 func (id *messageID) String() string {
-	return fmt.Sprintf("%d:%d:%d", id.ledgerID, id.entryID, id.partitionIdx)
+	return fmt.Sprintf("%d:%d:%d:%d", id.ledgerID, id.entryID, id.partitionIdx, id.batchIdx)
```

Comment:
if batchIndex = -1, do we still need to print it?

Reply:
Let's keep it consistent with the Java client. Don't print the batch index when it's -1.

Reply:
ok